Handle errors from regexp

thepetk opened the pull request:
## What does this PR do?

Replaces the `regexp.MustCompile` with `regexp.Compile` in case we provide a `var` and not a fixed/hardcoded value. This way it ensures that we will be able to catch any errors from regexp.

Example Replace:
```golang
re, err := regexp.MustCompile(regex)
```
With:
```golang
re, err := regexp.Compile(regex)
if err != nil {
   return ""
}
```

## Which issue(s) this PR fixes:

Fixes #199

yangcao77 reviewed (commented):
confirmed that the issue has been fixed with this change. 
can we add any test cases for this issue?

thepetk commented:
@yangcao77 I've added an extra test case in `alizer-test-resources` -> [this PR](https://github.com/devfile-resources/alizer-test-resources/pull/2) and I have updated the already existing [PR for tests update](https://github.com/redhat-developer/alizer/pull/198)

So this way we can merge the current workaround which will unblock has and this specific test case will be added with the next merge.

yangcao77 commented:
/lgtm
